src/Extension/CoreExtension.php

```diff
@@ -2055,4 +2066,39 @@ public static function parseAttributeFunction(Parser $parser, Node $fakeNode, $a
 
         return new GetAttrExpression($args[0], $args[1], $args[2] ?? null, Template::ANY_CALL, $line);
     }
+
+    private static function getPropertyReflector(object $object, string $property): \ReflectionProperty
+    {
+        $class = self::$classReflectors[$object::class] ??= new \ReflectionClass($object::class);
+
+        if ($class->hasProperty($property) && ($reflector = $class->getProperty($property))->isPublic()) {
+            return $reflector;
+        }
+
+        return new class ($property) extends \ReflectionProperty {
```

Comment:
if the array cast is only meant for dynamic properties, should we do a fast path for declared non-public properties ? Those names cannot be reused by a dynamic property.

Reply:
good idea, updated